pinot-query-planner/src/main/java/org/apache/pinot/query/planner/physical/DispatchablePlanVisitor.java

```diff
@@ -60,10 +60,17 @@ public QueryPlan constructDispatchablePlan(StageNode globalReceiverNode,
     dispatchablePlanContext.getDispatchablePlanStageRootMap().put(0, globalReceiverNode);
     // 3. add worker assignment after the dispatchable plan context is fulfilled after the visit.
     computeWorkerAssignment(globalReceiverNode, dispatchablePlanContext);
-    // 4. convert it into query plan.
+    // 4. compute the mailbox assignment for each stage.
+    computeMailboxAssignment(dispatchablePlanContext);
+    // 5. convert it into query plan.
```

Comment:
Let's add a TODO here for the 2 methods. later we want to refactor these into pluggable algorithms. (e.g. worker assignment visitor and mailbox assignment visitor)

but so far the entrypoint looks super clean and i love it

Reply:
added comments